### ISSUE TITLE
Volume snapshot does not work 

### ISSUE THREAD

**Issue by matofeder** [SovereignCloudStack/k8s-cluster-api-provider#408]
#### Desired state

As a KaaS user, I want to create a [volume-snapshot](https://kubernetes.io/docs/concepts/storage/volume-snapshots/) as is described e.g. [here](https://github.com/kubernetes/cloud-provider-openstack/blob/master/docs/cinder-csi-plugin/examples.md#snapshot-create-and-restore)

#### Current state

The volume snapshot feature is not working in the workload clusters managed by the latest (R4) version of k8s-cluster-api-provider (with the default settings, i.e. k8s version `v1.25.x`)

#### Steps to reproduce

1. Create a workload cluster by the [k8s-cluster-api-provider](https://github.com/SovereignCloudStack/k8s-cluster-api-provider) 
    - use default settings ( i.e. k8s version `v1.25.x`)
2. Create PVC and its snapshot (well described e.g. in [this example](https://github.com/kubernetes/cloud-provider-openstack/blob/master/docs/cinder-csi-plugin/examples.md#snapshot-create-and-restore))
3. K8s objects  `volumesnapshotcontents.snapshot.storage.k8s.io` and `volumesnapshots.snapshot.storage.k8s.io` are not created and volume snapshot is not created as well 

#### Reason

Currently, the script `apply_cindercsi.sh` wants to enable this feature with the deployment of related snapshot CRDs [here](https://github.com/SovereignCloudStack/k8s-cluster-api-provider/blob/main/terraform/files/bin/apply_cindercsi.sh#L22).

Since Kubernetes version `1.17`, the CSI external-snapshotter sidecar (sidecar deployed together with CSI cinder driver) controller has been split into two controllers: a [snapshot-controller](https://kubernetes-csi.github.io/docs/snapshot-controller.html) (an independent controller of any CSI Driver) and a CSI external-snapshotter sidecar (sidecar deployed together with CSI cinder driver).

Therefore the [snapshot-controller](https://kubernetes-csi.github.io/docs/snapshot-controller.html) should be deployed as well as part of k8s cluster management process to enable the volume snapshot feature in the k8s cluster. 

#### Posible solution 

Update the `apply_cindercsi.sh` script and deploy the [snapshot-controller](https://kubernetes-csi.github.io/docs/snapshot-controller.html) as is described e.g. [here](https://kubernetes-csi.github.io/docs/snapshot-controller.html#deployment)

**Comment by garloff** [comment]
Ideally, we should add a test case for this.
(The sonobuoy CNCF tests don't cover CSI by default unfortunately, otherwise we would have caught this one.)

**Comment by chess-knight** [comment]
> (The sonobuoy CNCF tests don't cover CSI by default unfortunately, otherwise we would have caught this one.)

@garloff what about `make check-storage` and `make check-csi`?

**Comment by garloff** [comment]
Would be good

**Comment by chess-knight** [comment]
But I mean that targets are actually there, see [check-storage](https://github.com/SovereignCloudStack/k8s-cluster-api-provider/blob/main/terraform/Makefile#L160) and [check-csi](https://github.com/SovereignCloudStack/k8s-cluster-api-provider/blob/main/terraform/Makefile#L163).

**Comment by chess-knight** [comment]
> OK, I had forgotten that we already have some good coverage with a specific sonobuoy check-csi target. So we should ensure they work (pass once we have the snapshots enabled) and then just start using them. THANKS!

Ok, I ran check-csi target(`sonobuoy.sh snapshotcluster --e2e-focus='CSI' --e2e-skip='Disruptive'`) 2 times successfully(two clusters with default configuration). Each run lasted 3+ hours, see e.g.:
```
=== Running sonobuoy conformance tests ... --e2e-focus=CSI --e2e-skip=Disruptive ===
...
=== Collecting results ===
...
Plugin: e2e
Status: passed
Total: 7069
Passed: 131
Failed: 0
Skipped: 6938

Plugin: systemd-logs
Status: passed
Total: 4
Passed: 4
Failed: 0
Skipped: 0

Run Details:
API Server version: v1.25.9
Node health: 4/4 (100%)
Pods health: 33/33 (100%)
...
=== Sonobuoy conformance tests passed in 11089s ===
```
I think that this issue can be closed as #415 is merged.
